### PR TITLE
Allow tracer components to be cleanly restarted after shutdown

### PR DESCRIPTION
**Pull request by marcotc**
We have inconsistent behaviour with `Configuration#shutdown!` where we shut down our components object (`        components.shutdown!`), but we keep its lingering instance in an instance variable (`@components`), with partially living references.

This prevented us from ever being able to restart the tracer, only ever returning dirty instances of the tracer on every call to `Datadog.tracer` after a shutdown.

This PR now allows the tracer components to be cleanly `shutdown!` and restarted, without lingering references.

### CODE REVIEW

File: lib/ddtrace/configuration.rb [hunk 1]
```diff
@@ -56,7 +56,10 @@ def logger
     end
 
     def shutdown!
-      components.shutdown! if instance_variable_defined?(:@components) && @components
+      if instance_variable_defined?(:@components) && @components
```
Comment (brettlangdon) [inline lib/ddtrace/configuration.rb:59]: should this change be in a different PR?

**Comment by ericmustin** [comment]
I think a commit slipped in here unrelated, otherwise lgtm and thanks for tracking this down, these flaky tests have been killing us

File: lib/ddtrace/configuration.rb (hunk 1 above)
Comment (marcotc) [reply lib/ddtrace/configuration.rb:59]: This is the important change, I updated the description of the PR to match.

**Comment by marcotc** [comment]
@ericmustin I removed the test fix, and kept the important change regarding tracer shutdown.

**Review by ericmustin — APPROVED** [review]
